Test member data

ondenman opened the pull request:
Tests scraper output against YAML file.
MemberPage#to_h should match the data specified in the YAML.

Part of everypolitician/everypolitician#600 (Ensure every scraper that has been ported to Scraped has a basic YAML test for at least one person)

This PR replaces https://github.com/everypolitician-scrapers/nigeria-national-assembly/pull/17

Commit:
Add scraper_test gem

Commit:
Add scraper_test task

Commit:
Configure travis to run tests

Commit:
Extract MemberPage to separate file

In preparation of a forthcoming commit, the class is being moved to its own
file so it can be required by scraper_test.

Commit:
Test member data

tmtmtmtm reviewed (changes requested):
@ondenman can I ask you to give this a better title? At the minute it's a bit of a word salad, which makes it quite hard to know what it's meant to be doing. Is this testing the "member data refactor"? Refactoring the "test member data"? Neither of them seem to match what's actually going on…

tmtmtmtm commented on lib/member_page.rb:
```diff
@@ -0,0 +1,89 @@
+
+  field :image do
+    image_url = box.css('.carousel-inner img/@src').text
+    image_url.include?('/avatar.jpg') ? nil : image_url
```
Would be good to make sure this is covered

ondenman replied:
Added in aacd30d

tmtmtmtm commented on lib/member_page.rb:
```diff
@@ -0,0 +1,89 @@
+  end
+
+  field :area do
+    constituency.empty? ? '' : [constituency, state].join(', ')
```
the 'empty' side of this doesn't seem to be covered

ondenman replied:
Added in 6763788

tmtmtmtm commented on lib/member_page.rb:
```diff
@@ -0,0 +1,89 @@
+    party_node.match(/^(.*)\s+\((.*)\)\s*$/) || abort("Bad party: #{party_node}")
+  end
+
+  TITLE_RE = /^(Hon\.|Sen\.) /
```
Both tests at the minute are for Hon., and none for Sen.

ondenman replied:
I've added the additional tests and will rebase to bring the branch up to date with master.

ondenman replied:
Added in 77bebab

tmtmtmtm reviewed (changes requested):
Generally looking good, but I think we need a few more test cases. In general we should make sure that any combination we explicitly code for is covered somewhere, to avoid any regressions later on those parts.

Commit:
Test member without profile image

Commit:
Test member without constituency

Commit:
Test member with Sen. prefix in source

The source page presents the member's name with a Sen. prefix.

The prefix should be removed.
